python/plugins/processing/algs/gdal/GdalUtils.py

```diff
@@ -518,12 +518,14 @@ def gdal_connection_details_from_layer(layer: QgsMapLayer) -> GdalConnectionDeta
 
     @staticmethod
     def ogrOutputLayerName(uri):
-        uri = uri.strip('"')
+        if uri.startswith('"') and uri.endswith('"'):
+            uri = uri.strip('"')
         return os.path.basename(os.path.splitext(uri)[0])
```

Comment:
This method just looks inherently unsafe to me. It's only used in one place (polygonize algorithm), and I'm not even certain it's required there (if we don't specify a layer name for gdal_polygonize, what happens? There's no circumstance in which we can use that algorithm to append a new layer to an existing file, given that the layer name is always auto-generated by ogrOutputLayerName to match the filename...)

Can we just remove this method accordingly?